### PR TITLE
feat(performance-metrics): Port vitalInfo - summary to metrics [INGEST-546]

### PR DESCRIPTION
**Pull request by priscilawebdev**
**This PR:**

- [x] Add metrics IF condition inside of the VitalInfo component, which renders the **Web Vitals** Widget. So if `isMetricsData` equals `true`, a metrics request will be made, otherwise an event request (this is currently under discussion and may be updated in a near future) 
- [x] Add several conditions to only add a query condition if `isMetricsData` equals `false`. That's because the metrics API does not support many of the tags,  throwing an error
- [x] Add Transaction Summary - Vital Widget Tests
- [x] Wrapped many tests with OrganizationContext, since MetricsSwitch uses `useOrganization()`

image

**Additional Information:**

As the Metrics API still cannot distinguish between frontend and non-frontend tags, the web vitals widget is only shown when the current project platform is of type frontend

### CODE REVIEW

File: static/app/views/performance/data.tsx [hunk 1]
```diff
@@ -450,7 +450,9 @@ function generateGenericPerformanceEventView(
   savedQuery.query = conditions.formatString();
 
   const eventView = EventView.fromNewQueryWithLocation(savedQuery, location);
-  eventView.additionalConditions.addFilterValues('event.type', ['transaction']);
+  if (!isMetricsData) {
```
Comment (JonasBa) [inline static/app/views/performance/data.tsx:453]: This seems to be doing some product/business related logic, so I would probably add a comment in all the places where we do something like this and on explaining why (unless it's super obvious and I just dont have enough knowledge yet 😅)

File: static/app/views/performance/data.tsx [hunk 2]
```diff
@@ -781,8 +793,8 @@ export function generatePerformanceEventView(
 }
 
 export function generatePerformanceVitalDetailView(
-  _organization: Organization,
-  location: Location
+  location: Location,
+  isMetricsData: boolean
```
Comment (JonasBa) [inline static/app/views/performance/data.tsx:797]: I usually try to avoid booleans like this because this "smells" like there could be multiple types in the future, in which case passing down `metricType: "metric data" | ...` can be a bit easier to extend. If you are sure that this wont change in the future tho, totally fine to leave as is :)

File: static/app/views/performance/landing/widgets/widgets/vitalWidgetMetrics.tsx [hunk 3]
```diff
@@ -50,6 +50,7 @@ export function VitalWidgetMetrics(props: PerformanceWidgetProps) {
   const [selectedListIndex, setSelectListIndex] = useState(0);
   const field = props.fields[0];
   const vital = settingToVital[chartSetting];
+  const orgSlug = organization.slug;
```
Comment (JonasBa) [inline static/app/views/performance/landing/widgets/widgets/vitalWidgetMetrics.tsx:53]: Probably fine to just pass organization.slug and keep the code more straightforward. Probably something to talk about at a TSC 🤔

File: static/app/views/performance/landing/widgets/widgets/vitalWidgetMetrics.tsx [hunk 4]
```diff
@@ -285,21 +294,27 @@ export function VitalWidgetMetrics(props: PerformanceWidgetProps) {
...
+      : response?.groups) ?? [];
 
   const vitalData: VitalData = {
     poor: 0,
     meh: 0,
     good: 0,
     p75: 0,
     ...groups.reduce((acc, group) => {
-      acc[group.by.measurement_rating] = group.totals[field];
+      acc[group.by.measurement_rating] = group.totals[field] ?? 0;
```
Comment (JonasBa) [inline static/app/views/performance/landing/widgets/widgets/vitalWidgetMetrics.tsx:317]: Is there a difference on the chart if we pass 0 vs undefined?

File: static/app/views/performance/transactionSummary/transactionEvents/index.tsx [hunk 5]
```diff
@@ -205,12 +205,23 @@ function getWebVital(location: Location): WebVital | undefined {
...
+  transactionName: string;
+  isMetricsData: boolean;
+}): EventView {
   const query = decodeScalar(location.query.query, '');
   const conditions = new MutableSearch(query);
-  conditions
-    .setFilterValues('event.type', ['transaction'])
-    .setFilterValues('transaction', [transactionName]);
+
+  if (!isMetricsData) {
```
Comment (JonasBa) [inline static/app/views/performance/transactionSummary/transactionEvents/index.tsx:220]: Same comment about the comment 😅

File: static/app/views/performance/vitalDetail/utils.tsx [hunk 6]
```diff
@@ -139,3 +140,10 @@ export function getMaxOfSeries(series: Series[]) {
   }
   return max;
 }
+
+export const vitalToMetricsField: Record<string, TransactionMetric> = {
```
Comment (JonasBa) [inline static/app/views/performance/vitalDetail/utils.tsx:144]: If you care about exact 1:1 mapping here, you could, but it may be an overkill here so feel free to dismiss this 
```
{ [K in keyof WebVital]: TransactionMetric[`SENTRY_TRANSACTIONS_MEASUREMENTS_${K}`] }
```

File: static/app/views/performance/vitalDetail/vitalInfo.tsx [hunk 7]
```diff
@@ -1,45 +1,105 @@
...
+  hideStates,
+  hideVitalPercentNames,
+  hideDurationDetail,
+  isMetricsData,
+}: Props) {
+  const api = useApi();
+
+  const vitals = Array.isArray(vital) ? vital : [vital];
+
+  function renderContent({
```
Comment (JonasBa) [inline static/app/views/performance/vitalDetail/vitalInfo.tsx:53]: This smells like it should either be inlined or split into a different component. I would say it's probably an antipattern to define renderContent fn for functional components and we should instead try to keep the render logic as concise as possible :) 
```
{({isLoading, vitalsData}) => isMetricData <Component1/> : <Component2/>
```

**Comment by priscilawebdev** [comment]
@JonasBa Thanks for your feedback.

We are introducing metrics in Performance and many of these checks will be removed or improved in the future. We currently display a switcher on the performance page (behind the feature flag `metrics-performance-ui`), where you can switch between metrics and transactions data... but in the future, we plan to remove this switcher, so the user will have an out-of-the-box experience

File: static/app/views/performance/vitalDetail/vitalInfo.tsx (hunk 7 above)
Comment (priscilawebdev) [reply static/app/views/performance/vitalDetail/vitalInfo.tsx:53]: sure thing... I've split this component in one of my open PRs but after discussing with the performance team, we decided to no fork widgets/components anymore as it can be hard to do maintenance, so as the content is basically the same, I thought that a reusable function was be a good option in this case

File: static/app/views/performance/landing/widgets/widgets/vitalWidgetMetrics.tsx (hunk 4 above)
Comment (priscilawebdev) [reply static/app/views/performance/landing/widgets/widgets/vitalWidgetMetrics.tsx:317]: yes... if I am not mistaken  in case some of the  `vitalData` property is `undefined` it will cause an error in another component , but i would have to double-check it

File: static/app/views/performance/landing/widgets/widgets/vitalWidgetMetrics.tsx (hunk 3 above)
Comment (priscilawebdev) [reply static/app/views/performance/landing/widgets/widgets/vitalWidgetMetrics.tsx:53]: when `organization.slug` is often repeated in the same file, I usually like to create a reusable variable .. but this is just a preference thing... we can for sure discuss this in the TSC meeting 😉

File: static/app/views/performance/data.tsx (hunk 2 above)
Comment (priscilawebdev) [reply static/app/views/performance/data.tsx:797]: I think the boolean is ok for now...but in the future, we can maybe have something like `dataType: 'metrics' | 'transaction'`

File: static/app/views/performance/data.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/views/performance/data.tsx:453]: yes sure... I'll add a comment 😉

File: static/app/views/performance/vitalDetail/utils.tsx (hunk 6 above)
Comment (priscilawebdev) [reply static/app/views/performance/vitalDetail/utils.tsx:144]: sure.. I could do something like:

```
Object.keys(
  WebVital
).reduce((acc, key) => {
  acc[WebVital[key]] = TransactionMetric[`SENTRY_TRANSACTIONS_MEASUREMENTS_${key}`];
  return acc;
}, {});
```

but I think I like more the current solution as it is more clear

**Review by JonasBa — APPROVED** [review]
🚢

File: src/sentry/sentry_metrics/__init__.py [hunk 8]
```diff
@@ -1,4 +1,4 @@
 """
 Module for the Metrics product built by sentry.
-Not to be confused with sentry.metrics, which collects statsd / datadog metrics
+Not to be confused with sentry.metrics, which collects stats / datadog metrics
```
Comment (Zylphrex) [inline src/sentry/sentry_metrics/__init__.py:3]: Was this change intentional? `statsd` isn't a typo as far as I know.

File: static/app/views/performance/vitalDetail/vitalInfo.tsx (hunk 7 above)
Comment (k-fish) [reply static/app/views/performance/vitalDetail/vitalInfo.tsx:53]: I agree with Jonas's point here, you making a new component definition on the fly basically, every time this FC gets called. You might be hitting some extra re-renders here.

**Review by k-fish — APPROVED** [review]

File: src/sentry/sentry_metrics/__init__.py (hunk 8 above)
Comment (priscilawebdev) [reply src/sentry/sentry_metrics/__init__.py:3]: oh, I thought it was a typo....will revert it...thanks!
